src/procedures/__tests__/WithdrawTaxes.ts

```diff
@@ -0,0 +1,216 @@
+
+  beforeEach(() => {
+    // Mock the context, wrappers, tokenFactory and dividendDistributionFactory to test WithdrawTaxes
+    contextMock = ImportMock.mockClass(contextModule, 'Context');
+    wrappersMock = ImportMock.mockClass(wrappersModule, 'PolymathBase');
+
+    tokenFactoryMock = ImportMock.mockClass(tokenFactoryModule, 'MockedTokenFactoryModule');
+    securityTokenMock = ImportMock.mockClass(contractWrappersModule, 'SecurityToken_3_0_0');
+    erc20DividendMock = ImportMock.mockClass(
+      contractWrappersModule,
+      'ERC20DividendCheckpoint_3_0_0'
+    );
+    ethDividendMock = ImportMock.mockClass(contractWrappersModule, 'EtherDividendCheckpoint_3_0_0');
+
+    dividendFactoryMock = ImportMock.mockClass(
+      dividendDistributionFactoryModule,
+      'DividendDistributionFactory'
+    );
+
+    tokenFactoryMock.mock(
+      'getSecurityTokenInstanceFromTicker',
+      securityTokenMock.getMockInstance()
+    );
+
+    contextMock.set('contractWrappers', wrappersMock.getMockInstance());
+    wrappersMock.set('tokenFactory', tokenFactoryMock.getMockInstance());
+
+    factoriesMockedSetup = mockFactories();
+    factoriesMockedSetup.dividendDistributionFactory = dividendFactoryMock.getMockInstance();
+    contextMock.set('factories', factoriesMockedSetup);
+
+    securityTokenGeneratedId = SecurityToken.generateId({
+      symbol: params.symbol,
+    });
+
+    target = new WithdrawTaxes(params, contextMock.getMockInstance());
+  });
+
+  afterEach(() => {
+    restore();
+  });
+
+  describe('Types', () => {
+    test('should extend procedure and have WithdrawTaxes type', async () => {
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe(ProcedureType.WithdrawTaxes);
+    });
+  });
+
+  describe('SetDividendsWallet', () => {
+    test('should throw if there is no valid security token being provided', async () => {
+      tokenFactoryMock
+        .mock('getSecurityTokenInstanceFromTicker')
+        .withArgs(params.symbol)
+        .throws();
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: `There is no Security Token with symbol ${params.symbol}`,
+        })
+      );
+    });
+
+    test('should throw if there is no valid dividend type being provided', async () => {
+      // Instantiate WithdrawTaxes with incorrect dividend type
+      target = new WithdrawTaxes(
+        { ...params, dividendType: 'wrong' as DividendType },
+        contextMock.getMockInstance()
+      );
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should add a transaction to the queue to withdraw tax withholdings from attached erc20 dividends distribution', async () => {
```

Comment:
"Attached" doesn't make sense here, since what is attached is the dividends module, not the distribution. Should just be "from erc20 dividends distribution"